shared/teams/join-team/join-from-invite.tsx

```diff
@@ -3,25 +3,34 @@ import * as React from 'react'
 import * as Container from '../../util/container'
 import * as Kb from '../../common-adapters'
 import * as Styles from '../../styles'
-import * as RPCTypes from '../../constants/types/rpc-gen'
 import * as Constants from '../../constants/teams'
 import {Success} from '.'
 
-type NoDetails = 'NODETAILS'
-const noDetails: NoDetails = 'NODETAILS'
-export type Props = Container.RouteProps<{details: RPCTypes.InviteLinkDetails | NoDetails}>
+const JoinFromInvite = () => {
+  const dispatch = Container.useDispatch()
+  const {inviteID: id, inviteKey: key, inviteDetails: details} = Container.useSelector(
+    state => state.teams.teamInviteDetails
+  )
+  const error = Container.useSelector(state => state.teams.errorInTeamJoin)
+  const loaded = details !== undefined || !!error
+  React.useEffect(() => {
+    if (loaded || key === '') {
+      return
+    }
+    if (key === '') {
+      // If we're missing the key, we want the user to paste the whole link again
```

Comment:
This feels a little bit like we're randomly applying a lot of extra duct tape to the joint where the invite link gets passed to the modal. Is there a case where we expect to get here without the link being plumbed?

Reply:
I'd like to get rid of it in the next PR, so that this flow is fully managed by the data layer. We only want to support two states:

1. If _anything_ is broken, we'll show a generic error screen and ask the user to re-enter the link (somewhat similar to the "Join team" flow). This includes:

    a. Disabled JavaScript (`keybase://team-invite-link/only-invite-id`), which would preview that invite.
    b. Invalid / expired link.
    c. No connectivity, perhaps?

2. Otherwise, we show the prompt.